### PR TITLE
Membership redirection

### LINKED ISSUE #1023: Restrict site by membership level, redirect to a WordPress page produces infinite loop
### Reproduction Steps

1. Create a Membership
2. on the Restrictions tab select Redirect to a "WordPress page" for Restricted Access Redirect
3. Select a WordPress page
4. leave the other settings as they are by default (Display a Message and Restricted Content Notice)
5. navigate through your dashboard -> LifterLMS (Settings) -> Membership
6. as Restrict site by membership level search and set the just created membership
7. save
9. as visitor, in front, navigate to whatever singular except the Membership created above (which is excluded by this restriction) and the selected WordPress page (which should be excluded as well)

### Expected Behavior

+ I would expect to be redirected to the selected WordPress page and see its content.

### Actual Behavior

+ I'm redirected to the selected WordPress page, which is still restricted, and then produces another redirect to itself and so on...

### Error Messages / Logs

n/a

### System Information

LifterLMS 3.37.9
WordPress 3.5.3

### Browser, Device, and Operating System Information

n/a

WIP: https://github.com/eri-trabiccolo/lifterlms/tree/membership-redirection
needs (unit) tests

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
## Description
Fixes #1023 
Also:
- added tests for sitewide membership restriction.
- fix cs/docs in includes/functions/llms.functions.access.php
- made sure "restriction notices" are print in wp pages too when redirected from a sitewide membership restriction.
- excluded privacy police page from sitewide restriction, see https://github.com/gocodebox/lifterlms/issues/1023#issuecomment-585839349

## How has this been tested?
manually, existing unit tests, new unit tests

## Screenshots
n/a 

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests.
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: includes/class.llms.template.loader.php [hunk 1]
```diff
@@ -313,6 +315,12 @@ public function restricted_by_membership( $info ) {
 			$msg      = '';
 			$redirect = '';
 
+			if ( 'yes' === $membership->get( 'restriction_add_notice' ) ) {
```
Comment (thomasplevy) [inline includes/class.llms.template.loader.php]: When evaluating yes/no values use `llms_parse_bool()`. Not a necessary change to make but using this method will allow greater consistency across the code base where yes/no values are stored. 

https://developer.lifterlms.com/reference/functions/llms_parse_bool/

Eg: `if ( llms_parse_bool( $membership->get( 'restriction_add_notice' ) ) ) {`

File: includes/functions/llms.functions.access.php [hunk 2]
```diff
@@ -1,24 +1,45 @@
...
+ *
+ * @since 3.0.0 Added the following functions:
+ *              `llms_is_page_restricted`, `llms_is_post_restricted_by_drip_settings`, `llms_is_post_restricted_by_prerequisite`,
+ *              `llms_is_post_restricted_by_time_period`, `llms_is_post_restricted_by_memberhsip`, `llms_is_post_restricted_by_sitewide_membership`
+ *
+ * @since 3.1.6 Added `llms_is_quiz_accessible` function.
+ * @since 3.2.4 Added `llms_get_restriction_message` function.
+ * @since 3.16.1 Unknown.
+ * @since 3.16.11 Unknown.
+ * @since 3.16.12 Unknown.
```
Comment (thomasplevy) [inline includes/functions/llms.functions.access.php]: duplicate version tag.

File: includes/functions/llms.functions.access.php [hunk 3]
```diff
@@ -255,7 +342,7 @@ function llms_is_post_restricted_by_drip_settings( $post_id, $user_id = null ) {
 	// if we're on a lesson, lesson id is the post id.
 	if ( 'lesson' === $post_type ) {
 		$lesson_id = $post_id;
-	} elseif ( 'llms_quiz' == $post_type ) {
+	} elseif ( 'llms_quiz' === $post_type ) {
```
Comment (thomasplevy) [inline includes/functions/llms.functions.access.php]: yes lol

File: includes/class.llms.template.loader.php (hunk 1 above)
Comment (eri-trabiccolo) [reply includes/class.llms.template.loader.php]: Yes, you're right, for some reason I keep on forgetting about that util.

Comment (thomasplevy) [reply includes/class.llms.template.loader.php]: No worries, just alerting you!

**Pull request merged**